pkg/messaging/nats/options.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package nats
+
+import (
+	"errors"
+
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+// ErrInvalidType is returned when the provided value is not of the expected type.
+var ErrInvalidType = errors.New("invalid type")
+
+// WithPrefix sets the prefix for the publisher.
+func WithPrefix(prefix *string) messaging.Option {
```

Comment:
With prefix for function names is typically used when extending some receiver value (usually a struct) consider renaming this